wdl-engine/src/eval/v1.rs

```diff
@@ -0,0 +1,2984 @@
+                    )),
+                }
+            }
+            None => Err(unknown_function(target.as_str(), target.span())),
+        }
+    }
+
+    /// Evaluates the type of an index expression.
+    fn evaluate_index_expr(&mut self, expr: &IndexExpr) -> Result<Value, Diagnostic> {
+        let (target, index) = expr.operands();
+        match self.evaluate_expr(&target)? {
+            Value::Compound(CompoundValue::Array(array)) => match self.evaluate_expr(&index)? {
+                Value::Primitive(PrimitiveValue::Integer(i)) => {
+                    match i.try_into().map(|i: usize| array.elements().get(i)) {
+                        Ok(Some(value)) => Ok(value.clone()),
+                        _ => Err(array_index_out_of_range(
+                            i,
+                            array.elements().len(),
+                            index.span(),
+                            target.span(),
+                        )),
+                    }
+                }
+                value => Err(index_type_mismatch(
+                    self.context.types(),
+                    PrimitiveTypeKind::Integer.into(),
+                    value.ty(),
+                    index.span(),
+                )),
+            },
+            Value::Compound(CompoundValue::Map(map)) => {
+                let ty = map.ty().as_compound().expect("type should be compound");
+                let key_type = match self.context.types().type_definition(ty.definition()) {
+                    CompoundTypeDef::Map(ty) => ty
+                        .key_type()
+                        .as_primitive()
+                        .expect("type should be primitive"),
+                    _ => panic!("expected a map type"),
+                };
+
+                match self.evaluate_expr(&index)? {
+                    Value::Primitive(i)
+                        if i.ty()
+                            .is_coercible_to(self.context.types(), &key_type.into()) =>
+                    {
+                        match map.elements().get(&i) {
+                            Some(value) => Ok(value.clone()),
+                            None => Err(map_key_not_found(index.span())),
+                        }
+                    }
+                    value => Err(index_type_mismatch(
+                        self.context.types(),
+                        key_type.into(),
+                        value.ty(),
+                        index.span(),
+                    )),
+                }
+            }
+            value => Err(cannot_index(
+                self.context.types(),
+                value.ty(),
+                target.span(),
+            )),
+        }
+    }
+
+    /// Evaluates the type of an access expression.
+    fn evaluate_access_expr(&mut self, expr: &AccessExpr) -> Result<Value, Diagnostic> {
+        let (target, name) = expr.operands();
+
+        // TODO: implement support for task values (required for WDL 1.2 support)
```

Comment:
Just a random question: do you keep track of these separately? Or are you planning to finish the current prototype and then you'll search the codebase for `TODO`?

Reply:
I'm keeping track of these separately, but also when the task evaluation is implemented, a bunch of these will be as well.